modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,414 @@
+// Module included in the following assemblies:
+//
+// * architecture/admission-controllers.adoc
+
+[id="admission-webhook-create_{context}"]
+= Create an admission webhook
+
+The steps below illustrate how to build a secure and portable webhook admission server.
+
+In this example, the webhook server is also an aggregated API server. This allows other OpenShift components to communicate with the webhook using internal credentials and facilitates testing using the `oc` command. Deploying an aggregated API server in this way also enables role based access control into the webhook, as well as preventing token information from other API servers being disclosed to the webhook. The API server will call `kube-apiserver` with its fully authenticated service account. The aggregated API server then calls to the webhook using the request header certificate. The webhook verifies incoming client certificates through the published request header aggregation CA.
+
+. Create a local CA key and certificate, plus a webhook server key and certificate which are signed by the local CA (replace `<variables>` accordingly):
++
+----
+mkdir _output/tmp/certs
+----
++
+----
+openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout "_output/tmp/certs/serving-ca.key" -out "_output/tmp/certs/serving-ca.crt" -subj "/C=<country>/ST=<state>/L=<city>/O=<company_name>/OU=<organizational_unit>/CN=ca/emailAddress=<email_address>/"
+----
++
+----
+echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","server auth"]}}}' > "_output/tmp/certs/serving-ca-config.json"
+----
+
+. Create a server key and certificate for the webhook:
```

Comment:
I have revampted those steps to use `openssl` only, which was a useful exercise for me because it furthered my understanding. I'm happy to remove those entirely if that is the right way to go in the doc. Can we reference KCS articles, because the steps are outlined clearly in https://access.redhat.com/solutions/15497?

Reply:
As part of the change towards a more generic high level procedure, I have now reduced the certificate generation and signing steps to one line.